src/runtime/pythonengine.cs

```diff
@@ -567,6 +576,60 @@ public static void Exec(string code, IntPtr? globals = null, IntPtr? locals = nu
             }
         }
 
+        /// <summary>
+        /// Gets the native thread ID.
+        /// </summary>
+        /// <returns>The native thread ID.</returns>
+        public static ulong GetNativeThreadID()
+        {
+            if (Runtime.PyVersion >= new Version(3, 8))
+            {
+                dynamic threading = Py.Import("threading");
+                return threading.get_native_id();
```

Comment:
Should get the `GIL`, I think.

Reply:
Isn't the user supposed to do that when calling the method, as I did in the test?

Reply:
This does not seem to be working on Ubuntu. Are you sure that this should work? Looking in Cpython tests I see that get_ident is used.